app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,14 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      if @identity&.user.present?
+      @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
```

Comment:
Add comment explaining this:
`User U has Provider P linked to U. U has email E1 while P has email E2. So, User table can't find E2 provided from auth hash, hence U is found by the user of identity having E2 as email`

Reply:
You can place this comment exactly too

Reply:
I added comments, what do you think?